frontend/src/stores/QueueStore.js

```diff
@@ -1,33 +1,101 @@
 import { observable, action, flow } from "mobx"
+import { computed } from "mobx"
 import { createContext } from "react"
+import moment from "moment"
 import api from "../api"
 
 export class QueueStore {
   constructor(rootStore) {
     this.rootStore = rootStore
   }
 
-  queueIds = {
-    stepQueue: 1,
-    legalServiceQueue: 2,
-    needleExchangeQueue: 3,
+  @observable queues = {
+    1: [],
+    2: [],
+    3: [],
+    4: [],
+    5: [],
+    6: [],
+    7: [],
+    8: [],
+    9: [],
   }
 
-  @observable stepQueue = []
-  @observable legalServiceQueue = []
-  @observable needleExchangeQueue = []
+  //Set tab order here
+  @computed get queueStats() {
+    return {
+      1: {
+        name: "TESTING",
+        length: this.queues[1].length,
+        waitTime: this.longestWait(this.queues[1]),
+      },
+      2: {
+        name: "CM",
+        length: this.queues[2].length,
+        waitTime: this.longestWait(this.queues[2]),
+      },
+      3: {
+        name: "SSHP",
+        length: this.queues[3].length,
+        waitTime: this.longestWait(this.queues[3]),
+      },
+      4: {
+        name: "LEGAL",
+        length: this.queues[4].length,
+        waitTime: this.longestWait(this.queues[4]),
+      },
+      5: {
+        name: "CRAFT",
+        length: this.queues[5].length,
+        waitTime: this.longestWait(this.queues[5]),
+      },
+      6: {
+        name: "PHAN",
+        length: this.queues[6].length,
+        waitTime: this.longestWait(this.queues[6]),
+      },
+      7: {
+        name: "STEP",
+        length: this.queues[7].length,
+        waitTime: this.longestWait(this.queues[7]),
+      },
+      8: {
+        name: "BIENSTAR",
+        length: this.queues[8].length,
+        waitTime: this.longestWait(this.queues[8]),
+      },
+      9: {
+        name: "SKWC",
+        length: this.queues[9].length,
+        waitTime: this.longestWait(this.queues[9]),
+      },
+    }
+  }
 
   @action
-  setQueue(queue, data) {
-    queue = data
+  setQueue(queueIndex, data) {
+    this.queues[queueIndex] = data
+  }
+
+  //Return the longest wait time in minutes
+  longestWait(queue) {
+    if (queue && queue.length > 0) {
+      return Math.floor(
+        (moment() - Math.min(...queue.map(x => moment(x.status.created_at)))) /
+          60000
+      )
+    } else {
+      return 0
+    }
   }
 
-  updateQueue = flow(function*(queue) {
-    const { ok, data } = yield api.getQueue(this.queueIds["queue"])
+  updateQueue = flow(function*(queueIndex) {
```

Comment:
should this really be called updateQueue?  It seems to just get whats in the db and set it in the store.  Maybe just getQueue?  I dunno, maybe your naming convention is right if you are thinking of it from a front end perspective, it is updating the queue state param on the front... regardless we should probably figure out what the naming convention should be because I find this confusing haha

Reply:
If it's confusing, then there's probably a better name.  I thought of it as something that should be called when the queues should be updated. Admittedly, it's being called more than that.  See my comment about using the hash with useEffect

Reply:
I changed it to getQueue

Reply:
great